prov/rxm/src/rxm_conn.c

```diff
@@ -87,6 +87,8 @@ static void rxm_close_conn(struct rxm_conn *conn)
 	}
 	fi_close(&conn->msg_ep->fid);
 	rxm_flush_msg_cq(conn->ep);
+	if(!dlist_empty(&conn->loopback_entry))
+		dlist_remove(&conn->loopback_entry);
```

Comment:
This looks iffy... if the entry is not empty then remove it? I think the entry would still come back as empty if it's the last one in the list so if the connections were freed in the opposite order that they were added, I think there could still be issues.
Also there's a missing space after the if :P

Reply:
when the head is initialized, `head->next = head`. when a new item is inserted, `item->next = head->next` so that the tail->next should always point to head.

Reply:
oh, and empty is defined by `head->next == head`, in this case, `head` is `&conn->loopback_entry`  It's checking to see if it's pointing to itself

Reply:
if the entry is still in the list (ep->loopback_list), the dlist_empty() test can't be true because the entry should at least be linked to loopback_list.

Reply:
on the other hand, the check is probably redundant. Does dlist_remove has any bad side effect if the list is empty?

Reply:
> if the entry is still in the list (ep->loopback_list), the dlist_empty() test can't be true because the entry should at least be linked to loopback_list.

right, this is what I'm trying to check. If it's not empty, that means it's in the loopback_list, and I want to remove it. 

> on the other hand, the check is probably redundant. Does dlist_remove has any bad side effect if the list is empty?

I think no, we likely don't need to check it and can just remove it.  if it's not in the list the remove would essentially do nothing

Reply:
most recent changes should address concerns discussed here